src/lib.rs

```diff
@@ -374,6 +374,8 @@ before taking old, removed peers offline.
 #![recursion_limit = "128"]
 // This is necessary to support prost and rust-protobuf at the same time.
 #![allow(clippy::identity_conversion)]
+// This lint recommends some bad choices sometimes.
```

Comment:
Is there an issue to reference here?

Reply:
split off into https://github.com/pingcap/raft-rs/pull/282

Reply:
They're still part of this PR so #282 blocks this :(

Reply:
I put them in a separate commit so it can still be reviewed. Just needs a rebase now.